src/worker/index.js

```diff
@@ -18,9 +23,38 @@ export async function handleRequest(event, context, fallback) {
   }
 
   try {
-    if (pathname.startsWith("/_flareact/props")) {
-      const pagePath = pathname.replace(/\/_flareact\/props|\.json/g, "");
+    const pagePath = pathname.replace(/\/_flareact\/props|\.json/g, "");
+
+    const normalizedPathname = normalizePathname(pagePath);
+    const resolvedPage = resolvePagePath(normalizedPathname, context.keys());
+    const resolvedPagePath = resolvedPage ? resolvedPage.pagePath : null;
 
+    if (config && typeof config.redirects !== "undefined") {
+      const reducedRedirect = config.redirects.find(
+        (item) => item.source === normalizedPathname || item.source === resolvedPagePath
+      );
+      if (reducedRedirect) {
+        const statusCode = reducedRedirect.permanent
+          ? PERMANENT_REDIRECT_STATUS
+          : TEMPORARY_REDIRECT_STATUS;
+        let destination = reducedRedirect.destination;
+
+        if (resolvedPage && resolvedPage.params) {
+          for (const param in resolvedPage.params) {
+            const regex = new RegExp(`\\[${param}\\]`);
+
+            destination = destination.replace(regex, resolvedPage.params[param]);
```

Comment:
Why this is needed? You're doing hard navigation with a user provided path. The final url will resolve on load anyways. 
Why not just `return Response.redirect(destinationURL, statusCode)`?

Reply:
If a request comes in at /blog/123 and the provided redirect is
```json
{
  "source": "/blog/[slug]",
  "destination": "/news/[slug]",
  "permanent": true
}
```
then it has to resolve the original page to match it with /blog/[slug] and to get the parameter to insert into the redirected URL (/news/123).

The response is being built manually because Response.redirect requires the full URL to be provided, not just a path.

Reply:
Oh, I see, its meant to be dynamic. Thought we're talking about hardcoded redirects. Great work

Reply:
Yeah, I've just gone back and looked at the original comment and realised I never mentioned dynamic redirects. Thanks!